### ISSUE TITLE
v1.4 incompatible with MacOs BigSur 

### ISSUE THREAD

**Issue by clamattu** [Benjamin-Dobell/Heimdall#490]
I've just tried to install it but it is considered too old to be installed by the system (v.11.5.2)

**Comment by amo13** [comment]
Did you try using the binary directly?

**Comment by clamattu** [comment]
Yes from [Downloads](https://glassechidna.com.au/heimdall/#downloads)

**Comment by amo13** [comment]
Then what do you mean by installing? Try putting the binary heimdall file (no .app or .pkg or similar) into a folder and calling it from a terminal (e.g. `./heimdall print-pit`).
Also, the version available from the downloads link you mentioned is somewhat outdated and known to cause trouble on newer macos versions and newer devices (unfortunately, this project has been abandoned for a while). Try the binary that I compiled with a bug fix from a pull request, available [here](https://heimdall.free-droid.com/).

**Comment by clamattu** [comment]
Ok, i have downladed, rebooted and tried it.
This is the result:

heimdall print-pit
```
Heimdall v1.4.2

Copyright (c) 2010-2017 Benjamin Dobell, Glass Echidna
http://www.glassechidna.com.au/

This software is provided free of charge. Copying and redistribution is
encouraged.

If you appreciate this software and you would like to support future
development please consider donating:
http://www.glassechidna.com.au/donate/

Initialising connection...
Detecting device...
ERROR: Failed to detect compatible download-mode device.
```

**Comment by amo13** [comment]
The last line states that no device in download mode was detected. Have you rebootet your device to download mode?

**Comment by clamattu** [comment]
Yes, it is with 
fap lock:off
Screen "Downloading... do not turn off target"

I have already changed few port of my MacMini

**Comment by amo13** [comment]
I merged a few overdue pull requests from this repo today. Can you please try with a freshly compiled version from [here](https://github.com/amo13/Heimdall/suites/3594666517/artifacts/86623046)?

**Comment by clamattu** [comment]
/Applications/bin/heimdall print-pit
```
Heimdall v1.4.2

Copyright (c) 2010-2017 Benjamin Dobell, Glass Echidna
http://www.glassechidna.com.au/

This software is provided free of charge. Copying and redistribution is
encouraged.

If you appreciate this software and you would like to support future
development please consider donating:
http://www.glassechidna.com.au/donate/

Initialising connection...
Detecting device...
ERROR: Failed to detect compatible download-mode device.
```

With a new cable that works on Win and Mac in non download-mode

Other suggestions?

**Comment by amo13** [comment]
What device is this?

**Comment by amo13** [comment]
@clamattu did you resolve the issue in the meantime?

**Comment by Grimler91** [comment]
This fork: https://gitlab.e.foundation/e/tools/heimdall/-/tree/new_master supposedly works with more recent mac versions, AFAIK that is the one that the LineageOS project built their mac binaries from

**Comment by amo13** [comment]
I'd rather have expected the other way around since /e/ is entirely relying on lineageos. How do you know? Did I miss a hint somewhere on lineage's site?

Anyway, the only additional commit to heimdall from the/e/ team seems to be only a fix for compilation issues on recent macos. It is pretty much the same as suggested in #468.

**Comment by Grimler91** [comment]
> How do you know? Did I miss a hint somewhere on lineage's site?

Hm, no, I might be completely wrong, I think I based it solely on the discussion in [one of the other threads](https://github.com/Benjamin-Dobell/Heimdall/issues/482#issuecomment-830796009).

> Anyway, the only additional commit to heimdall from the/e/ team seems to be only a fix for compilation issues on recent macos. It is pretty much the same as suggested in #468.

The /e/ changes look a bit cleaner than the ones in #468 (`if(APPLE)` seems better than `if(NOT MINGW)` for example), but you are right, pretty equivalent.

For reference, I have merged some of the PRs and made some minor changes over at [sourcehut](https://git.sr.ht/~grimler/Heimdall), but nothing to fix macos compilation (I don't own a mac)

**Comment by amo13** [comment]
It is way cleaner, you are right. And it also solves the compilation issues I had in [my fork](https://github.com/amo13/Heimdall). Thank you for the hint, @Grimler91 !

**Comment by danielboston38** [comment]
sorry for opening a closed issue but how do I get the binary? the option of opening a packages contents isn't in the menu anymore and I don't know how to do it.
